static/js/browser2/out_arcs.tsx

```diff
@@ -0,0 +1,157 @@
+ */
+
+import React from "react";
+import axios from "axios";
+import _ from "lodash";
+
+const ignoredOutArcProperties = new Set([
+  "provenance",
+  "kmlCoordinates",
+  "geoJsonCoordinates",
+  "geoJsonCoordinatesDP1",
+  "geoJsonCoordinatesDP2",
+  "geoJsonCoordinatesDP3",
+]);
+const maxNameWithLanguageValues = 10;
+
+interface OutArcsSectionPropType {
+  dcid: string;
+  outArcLabels: Array<string>;
+  provDomain: { [key: string]: URL };
+}
+
+interface OutArcsSectionStateType {
+  //TODO (chejennifer): replace the any type with an actual type
+  arcValuesData: { [predicate: string]: Array<any> };
+  propertyLabels: Array<string>;
+}
+
+export class OutArcsSection extends React.Component<
+  OutArcsSectionPropType,
+  OutArcsSectionStateType
+> {
+  constructor(props: OutArcsSectionPropType) {
+    super(props);
+    this.state = {
+      arcValuesData: {},
+      propertyLabels: [],
+    };
+  }
+
+  componentDidMount(): void {
+    this.fetchData();
+  }
+
+  private fetchData(): void {
+    const propValuesPromises = this.props.outArcLabels.map((label) => {
+      if (!ignoredOutArcProperties.has(label)) {
+        return axios
+          .get(`/api/browser/propvals/${label}/${this.props.dcid}`)
+          .then((resp) => resp.data);
+      }
+    });
+    Promise.all(propValuesPromises).then((propValuesData) => {
+      const outArcsByPredicate = {};
+      const propertyLabels = [];
+      propertyLabels.push("typeOf");
+      propValuesData.forEach((valuesData) => {
+        if (!valuesData || _.isEmpty(valuesData.values)) {
+          return;
+        }
+        const predicate = valuesData.property;
+        let values = valuesData.values.out;
+        if (predicate === "nameWithLanguage") {
+          const extra = values.length - maxNameWithLanguageValues;
+          values = _.slice(values, 0, maxNameWithLanguageValues);
+          values.push({
+            value: "(... " + extra + " more ...)",
+          });
+        }
+        outArcsByPredicate[predicate] = values;
+        if (predicate !== "typeOf") {
+          propertyLabels.push(predicate);
+        }
+      });
+      outArcsByPredicate["dcid"] = [{ value: this.props.dcid }];
+      propertyLabels.push("dcid");
+      this.setState({
+        arcValuesData: outArcsByPredicate,
+        propertyLabels: propertyLabels,
+      });
+    });
+  }
+
+  render(): JSX.Element {
+    if (_.isEmpty(this.state.propertyLabels)) {
+      return null;
+    }
+    return (
+      <div>
+        <table className="node-table">
+          <tbody>
+            <tr key="header">
+              <td className="property-column" width="25%">
+                <strong>Property</strong>
+              </td>
+              <td width="50%">
+                <strong>Value</strong>
+              </td>
+              <td width="25%">
+                <strong>Provenance</strong>
+              </td>
+            </tr>
+            {this.state.propertyLabels.map((propertyLabel) => {
```

Comment:
can make the row another component in this file. Now the deep indention makes it hard to read :)

Reply:
done